### PR TITLE
[docs] Explain how to define custom lldb commands

### PR DESCRIPTION
**Pull request by vedantk**

### CODE REVIEW

**Comment by mikeash** [comment]
Looks good!

**Review by mikeash — APPROVED** [review]

File: docs/DebuggingTheCompiler.rst [hunk 1]
```diff
@@ -377,6 +377,46 @@ Then by running ``lldb test -s test.lldb``, lldb will:
...
+    <debugger steps to the next instruction>
+
+Printing registers and single-stepping are by no means the only things you can
+do with custom commands. The LLDB Python API surfaces a lot of useful
+functionality, such as arbitrary expression evaluation.
+
+There are some pre-defined custom commands which can be especially useful while
+debugging the swift compiler. These commands live in
+``swift/utils/lldb/lldbToolBox.py``. There is a wrapper script available in
+``SWIFT_BINAR_DIR/bin/lldb-with-tools`` which launches lldb with those commands
```
Comment (jpsim) [inline docs/DebuggingTheCompiler.rst]: Is this a typo? Should it be `SWIFT_BINARY_DIR`?

Comment (vedantk) [reply docs/DebuggingTheCompiler.rst]: Thanks, yep, I'll fix it up.

**Comment by vedantk** [comment]
@swift-ci smoke test and merge

**Comment by vedantk** [comment]
@swift-ci smoke test and merge

**Pull request merged**